kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -1129,18 +1128,10 @@ func (c *ipamController) nodeIsBeingMigrated(name string) (bool, error) {
 		return false, err
 	}
 	// Get node to inspect labels
-	obj, ok, err := c.nodeIndexer.GetByKey(kname)
-	if !ok {
-		// Node doesn't exist, so isn't being migrated.
-		return false, nil
-	}
+	node, err := c.nodeLister.Get(kname)
```

Comment:
if you change the indexer by the lister, you should replace the !ok by err == ErrNotFound

Reply:
or you will fail now if the object is not found, that is an important behavior change

Reply:
Agreed, we need to handle all three cases - node exists, node doesn't exist, and an error when determining existence.